test/SparqlExpressionTest.cpp

```diff
@@ -398,17 +398,26 @@ auto testUnaryExpression =
                          liftVector(operand));
     };
 
-// Test `YearExpression`, `MonthExpression`, and `DayExpression`.
+// Test `YearExpression`, `MonthExpression`, `DayExpression`,
+//  `HoursExpression`, `MinutesExpression`, and `SecondsExpression`.
 TEST(SparqlExpression, dateOperators) {
   // Helper function that asserts that the date operators give the expected
   // result on the given date.
   auto checkYear = std::bind_front(testUnaryExpression, &makeYearExpression);
   auto checkMonth = std::bind_front(testUnaryExpression, &makeMonthExpression);
   auto checkDay = std::bind_front(testUnaryExpression, &makeDayExpression);
-  auto check = [&checkYear, &checkMonth, &checkDay](
-                   const DateOrLargeYear& date, std::optional<int> expectedYear,
+  auto checkHours = std::bind_front(testUnaryExpression, &makeHoursExpression);
+  auto checkMinutes = std::bind_front(testUnaryExpression, &makeMinutesExpression);
+  auto checkSeconds = std::bind_front(testUnaryExpression, &makeSecondsExpression);
+  auto check =
+      [&checkYear, &checkMonth, &checkDay, &checkHours, &checkMinutes, &checkSeconds](
+                   const DateOrLargeYear& date,
+                   std::optional<int> expectedYear,
                    std::optional<int> expectedMonth,
                    std::optional<int> expectedDay,
+                   std::optional<int> expectedHours,
+                   std::optional<int> expectedMinutes,
+                   std::optional<double> expectedSeconds,
```

Comment:
I think you can default all of them to ` = std::nullopt` (once also minutes and seconds return undefined consistently), then you can get rid of a lot of redundant arguments below (all the trailing `nullopt` arguments in the calls to `check`.

Reply:
Good morning :) 

Thank you for the feedback!  Both suggestions make perfect sense to me! :) I implemented them more or less the way you suggested. For the check if a given date contains time information, I implemented the new function `bool Date::hasTime()` (Date.h:308) because I assumed this test might be useful in other parts of the code again. 

What do you think? :)